tests/Seeds.ts

```diff
@@ -0,0 +1,489 @@
+import * as moment from 'moment';
+import { UserAccessType } from '../types';
+import { DatabaseConnection, EventFactory, MerchFactory, PortalState, UserFactory } from './data';
+
+function roundToHalfHour(date: moment.Moment): Date {
+  const HALF_HOUR_IN_MILLISECONDS = moment.duration(30, 'minutes').asMilliseconds();
+  return new Date(Math.round(date.valueOf() / HALF_HOUR_IN_MILLISECONDS) * HALF_HOUR_IN_MILLISECONDS);
+}
+
+function getGraduationYear(n: number) {
+  return moment().year() + n;
+}
+
+/**
+ * Our goal is to define some small system from which most actions are testable,
+ * e.g. checking into events (past/future, already-attended, ongoing), ordering
+ * merch (out of stock, insufficient credits, lifetime limit, successful), getting
+ * a decent-sized sliding leaderboard.
+ *
+ * Users: an admin, a couple committees (Hack, AI) and their accompanying
+ * staff, and a handful of members.
+ *
+ * Events: some staffed committee workshops, unstaffed general/committe socials,
+ * and staffed industry events, in the past, present, and future.
+ *
+ * There're also some "edge cases" here and there inspired by real data, e.g.
+ * admins granting bonus points to members that forgot/were unable to check-in.
+ */
+async function seed(): Promise<void> {
+  await DatabaseConnection.clear();
+
+  const ADMIN = UserFactory.fake({
+    email: 'acm@ucsd.edu',
+    accessType: UserAccessType.ADMIN,
+    firstName: 'ACM',
+    lastName: 'Admin',
+  });
+
+  const STAFF_GENERAL = UserFactory.fake({
+    email: 'stl005@ucsd.edu',
+    accessType: UserAccessType.STAFF,
+    firstName: 'Stanley',
+    lastName: 'Lee',
+  });
+  const STAFF_HACK = UserFactory.fake({
+    email: 'smhariha@ucsd.edu',
+    accessType: UserAccessType.STAFF,
+    firstName: 'Shravan',
+    lastName: 'Hariharan',
+  });
+  const STAFF_AI = UserFactory.fake({
+    email: 'stao@ucsd.edu',
+    accessType: UserAccessType.STAFF,
+    firstName: 'Stone',
+    lastName: 'Tao',
+  });
+
+  const MEMBER_FRESHMAN = UserFactory.fake({
+    email: 'ssteiner@ucsd.edu',
+    accessType: UserAccessType.STANDARD,
+    firstName: 'Steven',
+    lastName: 'Steiner',
+    graduationYear: getGraduationYear(4),
+  });
+  const MEMBER_SOPHOMORE = UserFactory.fake({
+    email: 'jpan@ucsd.edu',
+    accessType: UserAccessType.STANDARD,
+    firstName: 'Paul',
+    lastName: 'Pan',
+    graduationYear: getGraduationYear(3),
+  });
+  const MEMBER_JUNIOR = UserFactory.fake({
+    email: 'asudhart@ucsd.edu',
+    accessType: UserAccessType.STANDARD,
+    firstName: 'Andrea',
+    lastName: 'Sudharta',
+    graduationYear: getGraduationYear(2),
+  });
+  const MEMBER_SENIOR = UserFactory.fake({
+    email: 's3bansal@ucsd.edu',
+    accessType: UserAccessType.STANDARD,
+    firstName: 'Sumeet',
+    graduationYear: getGraduationYear(0),
+  });
```

Comment:
super nit: graduation years are going from 4 to 3 to 2 to 0, I think they should either be (3,2,1,0) or (4,3,2,1) for consistency (though I'm pretty sure it doesn't matter which)

Reply:
It's ordered (freshman, sophomore, junior, senior), so `getGraduationYear` takes years left and returns the current year minus years left.